xoto3/dynamodb/write_versioned/api2.py

```diff
@@ -168,3 +176,32 @@ def create_or_update_trans(vt: VersionedTransaction) -> VersionedTransaction:
         return table.put(creator_updater(table.get(key)(vt)))(vt)
 
     return create_or_update_trans
+
+
+def write_item(
+    table: TypedTable[T], writer: Callable[[Optional[T]], Optional[T]], key: ItemKey
+) -> TransactionBuilder:
+    """The most general purpose single-item-write abstraction, with
+    necessarily weak type constraints on the writer implementation.
+
+    Note that in DynamoDB parlance, a write can be either a Put or a
```

Comment:
Excellent. Yes. I see the value of sticking close to the Dynamo terminology. This makes that very clear. Thank you.